src/snappy/scala/io/iohk/ethereum/snappy/SnappyTest.scala

```diff
@@ -0,0 +1,83 @@
+package io.iohk.ethereum.snappy
+
+import io.iohk.ethereum.domain.{Block, Blockchain, Receipt}
+import io.iohk.ethereum.utils.Logger
+import org.scalatest.{FreeSpec, Matchers}
+
+import scala.concurrent.duration._
+
+class SnappyTest extends FreeSpec with Matchers with Logger {
+
+  val config = Config()
+  val pre = new Prerequisites(config)
+  import pre._
+
+  "Blockchain regression test" in {
+
+    val startN = targetBlockchain match {
+      case Some(tb) => config.startBlock.getOrElse(findHighestBlockNumber(tb) - 1).max(1)
+      case None => BigInt(1)
+    }
+    val targetN = config.targetBlock.getOrElse(findHighestBlockNumber(sourceBlockchain)).max(1)
```

Comment:
Minor: What about using `io.iohk.ethereum.db.storage.AppStateStorage#getBestBlockNumber` ?

Reply:
I would like to hold off with this change. The reason is I would need `sourceAppStateStorage` and `targetAppStateStorage` and it's starting to get messy. One of the changes planned for EC-303 refactor is to have the `Blockchain` be a single point of interaction with blockchain relevant data. That includes obtaining and modifying the best block number. For now I'm just gonna add a TODO comment, ok?

Reply:
Works for me :+1: